common/txmgr/types/client.go

```diff
@@ -56,3 +59,48 @@ type Blocks[BLOCK any, BLOCKHASH types.Hashable] interface {
 type Events[EVENT any, EVENTOPS any] interface {
 	FilterEvents(ctx context.Context, query EVENTOPS) ([]EVENT, error)
 }
+
+type TxmClient[
+	CHAIN_ID ID,
+	HEAD Head,
+	ADDR types.Hashable,
+	TX_HASH types.Hashable,
+	BLOCK_HASH types.Hashable,
+	R ChainReceipt[TX_HASH, BLOCK_HASH],
+	SEQ Sequence,
+	FEE Fee,
+	ADD any,
+] interface {
+	BatchSendTransactions(
+		ctx context.Context,
+		store TxStore[ADDR, CHAIN_ID, TX_HASH, BLOCK_HASH, R, SEQ, FEE, ADD],
+		attempts []TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD],
+		bathSize int,
+		lggr logger.Logger,
+	) ([]clienttypes.SendTxReturnCode, []error, error)
+	SendTransactionReturnCode(
+		ctx context.Context,
+		tx Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD],
+		attempt TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD],
+		lggr logger.Logger,
+	) (clienttypes.SendTxReturnCode, error)
+	PendingNonceAt(ctx context.Context, addr ADDR) (int64, error)
+	SequenceAt(ctx context.Context, addr ADDR, blockNum *big.Int) (SEQ, error)
+	BatchGetReceipts(
+		ctx context.Context,
+		attempts []TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD],
+	) (txReceipt []R, txErr []error, err error)
+	SendEmptyTransaction(
+		ctx context.Context,
+		txAttemptBuilder TxAttemptBuilder[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE, ADD],
+		seq SEQ,
+		gasLimit uint32,
+		fee FEE,
+		fromAddress ADDR,
+	) (txhash string, err error)
+	CallContract(
+		ctx context.Context,
+		attempt TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD],
+		blockNumber *big.Int,
```

Comment:
ohh, so eth client needs blocknumber as big.Int?
I see our head interface uses block number as int64. It is weird why it uses int64. Maybe sometime we can change it

Reply:
yeah there's still some weird type misalignment that will be addressed as we move more things around